test/integration/test/webpush_test.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const chakram = require('chakram'), expect = chakram.expect;
+
+var Prepper = require('../lib/testPrepper.js');
+
+Prepper.makeSuite('Test Push Service locally',function(){
+
+  before(Prepper.foxboxManager.foxboxLogin);
+
+  it('Check webpush service is registered',function(){  
+    var pushService = false;
+
+    return chakram.get(Prepper.foxboxManager.serviceListURL)
+    .then(function(listResponse) {
+      expect(listResponse).to.have.status(200);
+      for (var entry of listResponse.body) {
+        if (entry.adapter === 'webpush@link.mozilla.org') {
+          pushService = true;                
+        }
+      }
+      expect(pushService).equals(true);
+    });
+  });
+
+  it('Add push subscription',function(){     
+    var pushURI = Prepper.webPush_server.getEndpointURI();
+    var pushkey = Prepper.webPush_server.getPublicKey();
+    var setter = 'setter:subscribe.webpush@link.mozilla.org';
+    var getter =  'getter:subscription.webpush@link.mozilla.org';
+    var setterPayload = {
+      'select': {
+        'id':setter
+      }, 
+      'value': {
+        'Json': {
+          'subscriptions':[{
+            'push_uri':pushURI,
+            'public_key':pushkey
+          }]}}};
+          var getterPayload = {'id': getter};    
+
+          return chakram.put(Prepper.foxboxManager.setterURL,setterPayload)
+          .then(function(cmdResp){
+           expect(cmdResp).to.have.status(200);
+           expect(cmdResp.body[setter]).equals(null);
```

Comment:
Following up on the [null discussion](https://github.com/fxbox/foxbox/pull/395#discussion_r60411373): I think it would be good to specify why we check the null state in a comment